src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -245,14 +245,6 @@ private static async Task<RestoreResultPair> ExecuteAsync(RestoreSummaryRequest
             // Run the restore
             var request = summaryRequest.Request;
 
-            // Read the existing lock file, this is needed to support IsLocked=true
-            // This is done on the thread and not as part of creating the request due to
-            // how long it takes to load the lock file.
-            if (request.ExistingLockFile == null)
-            {
-                request.ExistingLockFile = LockFileUtilities.GetLockFile(request.LockFilePath, log);
-            }
-
```

Comment:
Does this change break the scenario described in the comment?

Reply:
To answer my own question:  nope, because `request.ExistingLockFile` will lazily load `request.LockFilePath` anyway.